LaunchDarkly/LaunchDarkly/LDClient.swift

```diff
@@ -327,6 +327,36 @@ public class LDClient {
         identifyQueue.enqueue(request: identifyTask)
     }
 
+    /**
+     Sets the LDContext into the LDClient inline with the behavior detailed on `LDClient.identify(context: completion:)`. Additionally,
+     this method will ensure the `completion` parameter will be called within the specified time interval.
+
+     Note that the `completion` method being invoked does not mean that the identify request has been cancelled. The identify request will
+     continue attempting to complete as it would with `LDClient.identify(context: completion:)`. Subsequent identify requests queued behind
+     a timed out request will remain blocked (or shed) until the in flight request completes.
+
+     - parameter context: The LDContext set with the desired context.
+     - parameter timeout: The upper time limit before the `completion` callback will be invoked.
+     - parameter completion: Closure called when the embedded `setOnlineIdentify` call completes, subject to throttling delays.
+     */
+    public func identify(context: LDContext, timeout: TimeInterval, completion: @escaping ((_ result: IdentifyResult) -> Void)) {
+        var cancel = false
+
+        DispatchQueue.global().asyncAfter(deadline: .now() + timeout) {
+            guard !cancel else { return }
+
+            cancel = true
+            completion(.timeout)
+        }
+
+        identify(context: context) { result in
+            guard !cancel else { return }
+
+            cancel = true
+            completion(result)
```

Comment:
Is it possible that both the timeout async call and this call get past the guard clause at the same time?

Reply:
I do not believe so. Both the timeout and the identify completion code rely on the global DispatchQueue. So the requests should always be appropriately ordered.